hw20/js/28students.js

```diff
@@ -0,0 +1,221 @@
+class Human {
+    name = null;
+    surname = null;
+    age = null;
+
+    constructor(person = { name: 'John', surname: 'Doe', age: 30 }) {
+        this.name = person.name;
+        this.surname = person.surname;
+        this.age = person.age;
+    }
+
+    getFullName() {
+        return `${this.name} ${this.surname}`;
+    }
+    setFullName(fullName) {
+        const [ name, surname ] = fullName.split(' ');
+
+        this.name = name;
+        this.surname = surname;
+    }
+}
+
+class Student extends Human {
+    marks = [];
+
+    constructor(person = { name: 'JohnS', surname: 'DoeS', age: 18, marks: [] }) {
+        super(person);
+
+        this.marks = person.marks;
+    }
+
+    getMarks() {
+        return this.marks;
+    }
+    getAverageMark() {
+        return this.marks.reduce((total, mark) => total + mark, 0) / this.marks.length;
+    }
+    getMinMark() {
+        return Math.min(...this.marks);
+    }
+    getMaxMark() {
+        return Math.max(...this.marks);
+    }
+}
+
+class FakeStudent extends Student {
+    #cheatedMarks = [];
+    #MAX_MARK = 12;
+    #MULTIPLIER = 2;
+
+    constructor(person = { name: 'JohnFS', surname: 'DoeFS', age: 24, marks: [] }) {
+        super(person);
+
+        this.#cheatedMarks = this.#cheat();
+    }
+
+    #cheat() {
+        return this.marks.reduce((cheatedMarks, mark) => {
+            const double = mark * this.#MULTIPLIER;
+            const target = double > this.#MAX_MARK ? this.#MAX_MARK : double;
+
+            return [...cheatedMarks, target];
+        }, [])
+    }
+    getMarks() {
+        return this.#cheatedMarks;
+    }
+    getAverageMark() {
+        return this.#cheatedMarks.reduce((total, mark) => total + mark, 0) / this.#cheatedMarks.length;
+    }
+    getMinMark() {
+        return Math.min(...this.#cheatedMarks);
+    }
+    getMaxMark() {
+        return Math.max(...this.#cheatedMarks);
+    }
+}
+
+class Teacher extends Human {
+    group = [];
```

Comment:
no need for this. You've already passed `group` in constructor

Reply:
```+```